### ISSUE TITLE
On each platform release: Update Flow version in Vaadin Docs Flow API link

### ISSUE THREAD

**Issue by Haprog** [vaadin/platform#523]
The problem is that in Vaadin Docs https://vaadin.com/docs for Vaadin 10+ it shows a link in the sidebar "Vaadin Flow / Flow API" which links to e.g. https://vaadin.com/api/com.vaadin/flow-server/1.0 (depending on which version of docs you are browsing).

If you leave the version out of the URL it will always redirect to latest version, if there is just `1` it will redirect to latest release of version `1`, if there is `1.0` it goes to latest of that (currently `1.2.4`). But we don't want to use these dynamic URLs here since then the target would be determined by latest releases of Flow even if that version of Flow is not yet included in any platform release. So we instead want to use the exact version in the link so that it links to the specific version of Flow that is included in the latest release of the selected major platform version.

These links are currently managed in:
- https://github.com/vaadin/vaadin-docs/blob/vaadin10/website/_data/docs.yml#L46
- https://github.com/vaadin/vaadin-docs/blob/vaadin11/website/_data/docs.yml#L46
- https://github.com/vaadin/vaadin-docs/blob/vaadin12/website/_data/docs.yml#L46

One branch for each "version of Docs"

When Vaadin 13 beta (and so on) is released there will probably also be 
- https://github.com/vaadin/vaadin-docs/blob/vaadin13/website/_data/docs.yml

Which should link to the Flow version that is included in the latest beta release of Vaadin 13 (until we have a first stable/GA version of it, after which this should always link to the version in latest stable Vaadin 13).

So basically for any `vaadin<version>` branch (where version >= 10) in `vaadin-docs` the link to Flow API in `website/_data/docs.yml` needs to be updated with the version number which should be the `flow` version that is included in the latest GA/stable release of that specific major platform version (unless there is no stable/GA version yet for that major, then use the latest prerelease version).

**Comment by qtdzz** [comment]
- [x] Created a build for this task.
- [x] Added to the release checklist.
- [x] Updated the platform team wiki.
